src/main/webapp/creatives/maps/simid_maps.js

```diff
@@ -1,8 +1,44 @@
 /**
  * A sample SIMID ad that shows a map of nearby locations
  */
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+}
+const DEFAULT_BUTTON_LABEL = "Location";
+
 class SimidMapCreative extends BaseSimidCreative {
   constructor() {
     super();
   }
+
+  /** @override */
+  onStart(eventData){
+    super.onStart(eventData);
+    const adParams = this.creativeData.adParameters;
+    const buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL]; 
+    if (buttonLabel == undefined) {
+      buttonLabel = DEFAULT_BUTTON_LABEL;
+    }
+    else {
+      buttonLabel = JSON.parse(adParams)[AdParamKeys.BUTTON_LABEL];
+    }
+    let searchQuery = JSON.parse(adParams)[AdParamKeys.SEARCH_QUERY];
+    //ToDo(juliareichel@): handle case where searchQuery is undefined
+    const marker = JSON.parse(adParams)[AdParamKeys.MARKER];;
+    this.specifyButtonFeatures_(buttonLabel);
+  }
+ 
+  /** @private */
```

Comment:
include description of what this function should do (the function below as well), and parameter types 

```
  /**
  * @param {string} buttonLabel description
  */
```

or if you use default params
```
/**
 * @param {string=} buttonLabel description of what buttonLabel is
 */
```

Reply:
added